model/labels/labels_stringlabels.go

```diff
@@ -527,48 +521,27 @@ func marshalLabelToSizedBuffer(m *Label, data []byte) int {
 	return len(data) - i
 }
 
-func sizeVarint(x uint64) (n int) {
-	// Most common case first
-	if x < 1<<7 {
+func sizeWhenEncoded(x uint64) (n int) {
+	if x < 255 {
 		return 1
+	} else if x <= 1<<24 {
+		return 4
 	}
-	if x >= 1<<56 {
-		return 9
-	}
-	if x >= 1<<28 {
-		x >>= 28
-		n = 4
-	}
-	if x >= 1<<14 {
-		x >>= 14
-		n += 2
-	}
-	if x >= 1<<7 {
-		n++
-	}
-	return n + 1
-}
-
-func encodeVarint(data []byte, offset int, v uint64) int {
-	offset -= sizeVarint(v)
-	base := offset
-	for v >= 1<<7 {
-		data[offset] = uint8(v&0x7f | 0x80)
-		v >>= 7
-		offset++
-	}
-	data[offset] = uint8(v)
-	return base
+	panic("String too long to encode as label.")
```

Comment:
It would be nice to revisit our validation on Prometheus ingestion, if we check this.

Reply:
Yeah.  The main validation functions are in `prometheus/common` library, but we could have extra validation on scrape and on remote-write into Prometheus.
For general common sense I think you'd want a limit defaulting lower than 16MB.

Reply:
* Created #16525 to track.